tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -232,7 +233,12 @@ static PyObject * THPVariable_numel(PyObject* self, PyObject* args)
    if (jit::tracer::isTracing()) {
      return wrap(jit::tracer::getNumelOf(self_));
    } else {
-     return THPUtils_packInt64(self_.numel());
+     auto si = self_.sym_numel();
+     if (si.is_symbolic()) {
+       return py::cast(si.toSymIntNodeImpl()).release().ptr();
+     } else {
+       return THPUtils_packInt64(si.data());
```

Comment:
I'm seeing this if-else branch a lot; we could simplify this by writing a type caster for SymInt; then you can just say `py::cast`

Reply:
would it be okay to push to another PR? This PR is already not small and adding a caster sounds like a general SymInt clean up?

Reply:
ok